mesh: Reset Heartbeat Publication timer before sending message

MariuszSkamra opened the pull request:
The timer should be reset before sending to ensure correct publication period.

This patch allows to pass MESH/NODE/CFG/HBP/BV-02-C.

This patch is ported from mynewt:
Commit: d4b84638df47e7ea21629e6919f547f5dcd47285

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra committed:
Bluetooth: mesh: Reset Heartbeat Pub timer before sending message

The timer should be reset before sending to ensure correct
publication period.

This patch allows to pass MESH/NODE/CFG/HBP/BV-02-C.

This patch is ported from mynewt:
Commit: d4b84638df47e7ea21629e6919f547f5dcd47285

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

jhedberg commented on subsys/bluetooth/host/mesh/cfg_srv.c:
```diff
@@ -3057,20 +3057,20 @@ static void hb_publish(struct k_work *work)
-	}
-
 	period_ms = hb_pwr2(cfg->hb_pub.period, 1) * 1000;
 	if (period_ms) {
```
One thing I'd consider adding (as a separate patch) is && cfg->hb_pub.count > 1 to this condition, so that we don't schedule the delayed work if this was the last heartbeat message.

jhedberg replied:
Actually, > 1 is not sufficient since you need to deal with 0xffff as well (which means "send heartbeat messages indefinitely")

jhedberg replied:
So actually extending the condition with "&& cfg->hb_pub.count != 1" would likely suffice

MariuszSkamra replied:
good catch!

jhedberg commented:
Copying here the reason CI rejected the PR:

> 3: UC4 Line exceeds max length (78>72): "The timer should be reset before sending to ensure correct publication period."

MariuszSkamra committed:
Bluetooth: mesh: Check Heartbeat Pub count prior scheduling delayed work

This check is needed to not schedule the delayed work if current
message to be sent is the last heartbeat message.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra commented:
- Updated

@michal-narajowski FYI
